langserver/handler.go

```diff
@@ -240,24 +247,33 @@ func (h *langHandler) linter() {
 		running[uri] = cancel
 
 		go func() {
-			diagnostics, err := h.lint(ctx, uri)
+			uriToDiagnostics, err := h.lint(ctx, uri)
 			if err != nil {
 				h.logger.Println(err)
 				return
 			}
 
-			if diagnostics == nil {
-				return
+			for diagURI, diagnostics := range uriToDiagnostics {
+				copiedDiagnostics := diagnostics
+				copiedDiagURI := diagURI
+				if diagURI == "file:" {
+					copiedDiagURI = uri
+				}
+				go func() {
+					version := 0
+					if _, ok := h.files[uri]; ok {
+						version = h.files[uri].Version
+					}
+					h.conn.Notify(
+						ctx,
+						"textDocument/publishDiagnostics",
+						&PublishDiagnosticsParams{
+							URI:         copiedDiagURI,
+							Diagnostics: copiedDiagnostics,
+							Version:     version,
+						})
+				}()
```

Comment:
Thanks your contribution. What reason to use goroutine?

Reply:
Thank you, I removed it :pray: